### PR TITLE
add development and testing guidelines

### PR DESCRIPTION
**Pull request by goru001**
1. Added example for how bot developers should go about debugging
2. Added guidelines for how bot developers should do bot development
3. Added guidelines for how bot developers should do testing

### CODE REVIEW

File: docs/bot-builder/advanced/testing-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    
+    Therefore, you should make sure **your bot is learning the right
+     things by testing it on phrases / words occuring repeatedly in your user says**
+     
+Example: If sentence is "Benefits of SIP", then test your bot on:
+    * Benefits of bvdfbv fjdb (Phrase + Gibberish/noise)
+    * cricket benefits (Gibberish/noise + Phrase)
+    * benefits (Phrase)
+    
+Solve any of the issues you encounter using debugging guidelines.
```
Comment (krupalmodi18) [inline docs/bot-builder/advanced/testing-guidelines.md]: Please add hyperlink to "debugging guidelines".

File: docs/bot-builder/advanced/development-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    * the sentence is `out of scope sentence` for the bot
+
+    *Don’t add these sentences to any of the user says*
+
+    This is in principle with doing Test-driven development (TDD),
+     which is a development technique where you must first write a test
+      that fails before doing any new development.
+
+    Once you have a reasonably sized Test set, you can move to the next step. Checkout the Testing Guidelines to see
+     how you can prepare test cases.
```
Comment (krupalmodi18) [inline docs/bot-builder/advanced/development-guidelines.md]: Add link to "Testing guidelines".

File: docs/bot-builder/advanced/development-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+> **_Pro Tip -_**
+> You should keep revisiting the test set during development to:
+> * Test whatever you’ve built works on these test cases
+> * Add new test cases, edge cases which you might discover during development
+
+
+2. **Development**
+
+    * Add user says, following user-says guidelines, preferably for max
```
Comment (krupalmodi18) [inline docs/bot-builder/advanced/development-guidelines.md]: Add link to user says guidelines.

File: docs/bot-builder/advanced/debugging-guidelines.md [hunk 4]
```diff
@@ -53,6 +53,47 @@ In any of the cases mentioned above, the following information / concepts might
 
 ![Debug Debug Chart](assets/debug_flow_chart.png)
 
+
```
Comment (krupalmodi18) [inline docs/bot-builder/advanced/debugging-guidelines.md]: A part from this, line number 19 -25 needs to be edited with better explanation.

File: docs/bot-builder/advanced/development-guidelines.md (hunk 2 above)
Comment (goru001) [reply docs/bot-builder/advanced/development-guidelines.md]: done.
